tools/roosttest_unit/test_iced_bundle_launch.py

```
@@ -0,0 +1,565 @@
+            any(arg.startswith("ROOST_BUNDLE_PROFILE=") for arg in open_argv),
+            f"ROOST_BUNDLE_PROFILE must never be forwarded into the bundle "
+            f"launch (it exists to exercise the bundle-id-derived default "
+            f"profile): {open_argv!r}",
+        )
+        assert_identity.assert_called_once()
+        assert_canary.assert_called_once_with("iced")
+
+    def _launch_argv_with_env(self, env: dict[str, str]) -> list[str]:
+        with (
+            patch.dict(os.environ, env, clear=True),
+            patch("ui._iced_bundle_ui_log_path", return_value=Path(tempfile.mktemp())),
```

Ruff (0.16.1)

[error] Use of insecure and deprecated function (`mktemp`)

(S306)